### DISCUSSION TITLE
Welcome to pulse Discussions!

### DISCUSSION THREAD

**Discussion by code-xhyun** [pulsecron/pulse#26] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by blankart** [comment]
May i know if it's safe to migrate from agenda to pulse without any data loss? I have an existing server with existing jobs that were scheduled months from now

**Reply by code-xhyun** [reply]
@blankart 
First, I need to know exactly which version of Agenda you are using. There are two versions: https://github.com/agenda/agenda and https://github.com/hokify/agenda, and they have some differences. If you could provide an example of the data currently stored, it would be more helpful for me.

P.S. Also, Pulse is an advanced version of Agenda, so it has some different functionalities.

**Reply by blankart** [reply]
Here's an example of one of the stored jobs:

```
{
  "_id": {
    "$oid": "66319edb3d7a578a0da3eaa0"
  },
  "name": "<REDACTED>",
  "data": {
    "id": "66319e5fd140c81c4bd2da80"
  },
  "priority": 0,
  "shouldSaveResult": false,
  "type": "normal",
  "nextRunAt": null,
  "lastModifiedBy": null,
  "lockedAt": null,
  "lastRunAt": {
    "$date": "2024-05-22T02:56:09.576Z"
  },
  "lastFinishedAt": {
    "$date": "2024-05-22T02:56:10.620Z"
  }
}
```

We're currently using `agenda@5.0.0`. We can adjust the current codebase to make it `pulse` compatible, as long as we can ensure that existing stored jobs created using agenda is guaranteed to run

**Reply by code-xhyun** [reply]
The following is an example of the jobs data in Pulse. Comparing this with Agenda, it appears to be the same, and there should be no issues. However, since I haven't seen the part of your code currently using Agenda, I can't guarantee this perfectly. Therefore, I strongly recommend you test it out.
```
{
  "_id": {
    "$oid": "664ae82720d4b2e571170a31"
  },
  "name": "send email",
  "data": {},
  "priority": 0,
  "shouldSaveResult": false,
  "attempts": 0,
  "backoff": null,
  "type": "normal",
  "nextRunAt": {
    "$date": "2024-05-21T08:50:22.641Z"
  },
  "repeatInterval": "1 minute",
  "repeatTimezone": null,
  "startDate": null,
  "endDate": null,
  "skipDays": null,
  "lastModifiedBy": null,
  "lockedAt": null,
  "lastRunAt": {
    "$date": "2024-05-21T08:49:22.641Z"
  },
  "runCount": 134,
  "finishedCount": 134,
  "lastFinishedAt": {
    "$date": "2024-05-21T08:49:22.642Z"
  }
}
```

**Reply by code-xhyun** [reply]
@blankart 
If there are any issues, please let me know, and I will create a new version that resolves the problem immediately.
